test/integration/client-side-operations-timeout/client_side_operations_timeout.prose.test.ts

```diff
@@ -1256,29 +1254,20 @@ describe('CSOT spec prose tests', function () {
           .catch(() => null);
         await internalClient.db('admin').command(failpoint);
 
-        const hello = await internalClient.db('admin').command({ hello: 1 });
-        maxBsonObjectSize = hello.maxBsonObjectSize;
-        maxMessageSizeBytes = hello.maxMessageSizeBytes;
-
         client = this.configuration.newClient({ timeoutMS: 2000, monitorCommands: true });
       });
 
-      it.skip('performs two bulkWrites which fail to complete before 2000 ms', async function () {
+      it('performs two bulkWrites which fail to complete before 2000 ms', async function () {
         const writes = [];
-        client.on('commandStarted', ev => writes.push(ev));
+        client.on('commandStarted', filterForCommands('bulkWrite', writes));
```

Comment:
Yes.  What scenario are you thinking we are missing?

The test specifically needs two bulk writes to be executed to test that when we execute multiple batches, timeoutMS applies to all batches collectively.  It doesn't matter if each bulk write results in a cursor and there are also getMores and killCursors.

Reply:
Does this test run over more than one batch? 

I am inspired by the unified tests 'ignoreExtraEvents' being something we opt-in to, and maybe here we are considering the test to be opt-ed in to ignoring the operations you call out, but if they're guaranteed to always be there and in a certain order I would assert them to catch any surprise changes. I originally thought the only operations was bulkWrite, but if those other operations are there then filtering seems fine.

Reply:
This test always runs two batches.  The test description is:

> performs two bulkWrites which fail to complete before 2000 ms

makeMultiBatchWrite always constructs a multi-batch `models` array.

There shouldn't be any getMores or killCursors sent for this test with the current server implementation but if there were, they do not matter.  Because the point of the test is that batch 1 execution time + batch 2 execution time <= timeoutMS.  getMores and killCursors are part of the total batch execution time

Reply:
> the point of the test is that batch 1 execution time + batch 2 execution time <= timeoutMS
>
> `// from the test title:`
> fail to complete before 2000 ms

IIUC this is not what we are asserting should we add a time-taken assertion? 

If those operations shouldn't happen then should we check that they don't? It could be surprising if that changes and it does seem more inline with the step "Verify that two `bulkWrite` commands were executed as part of the `MongoClient.bulkWrite` call.". It doesn't say "ignore extra commands" but it also does not say to assert they exist either, so I am not confident we should filter extras out.

Reply:
> IIUC this is not what we are asserting should we add a time-taken assertion?

Yes, this is what warren was mentioning this morning - no spec or prose test asserts on the durations of timeouts and that is problematic.  However, this test is aligned with all other CSOT prose tests.  The pattern is to set a failpoint to block the command you want to fail and assert it fails with a timeout error.  I also do not like deviating from the spec test because that only leads to confusion in the future.

I have a copy of this test that does assert on the timing in the node specific tests.  Does that satisfy what you're looking for?

> If those operations shouldn't happen then should we check that they don't? It could be surprising if that changes and it does seem more inline with the step "Verify that two bulkWrite commands were executed as part of the MongoClient.bulkWrite call.". It doesn't say "ignore extra commands" but it also does not say to assert they exist either, so I am not confident we should filter extras out.

When I say "shouldn't", I mean "they don't with server 8.0".  It doesn't matter either way to the test if they do or do not.

The implementation of the test matches the verbiage in the prose test and I am going to leave it as-is.

Reply:
Sure copying and checking time sounds fine. I brought up measuring time because you say "the point of the test [...] execution time <= timeoutMS" yet that is not what is covered by the steps in the prose nor in code, so where is that test intent coming from?

> Does that satisfy what you're looking for?

I was originally looking for confirmation that extra commands run truly do not matter since the test isn't clear on that and inspired by the unified runner's default, I figure the test would have explicitly stated if you should ignore extra events. Since it omits mention of ignoring extra commands I do not think we should be.

> they don't with server 8.0

Will this change with other server versions?

Reply:
From slack: we can filter the commands. We have deadline coverage in the other node test file